### DISCUSSION TITLE
Welcome to ZeroUI Discussions!

### DISCUSSION THREAD

**Discussion by dec0dOS** [dec0dOS/zero-ui#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by AxlRose1976** [comment]
Hi,

I am trying to run this docker image on arm.
But it doesn´t let me pull the image.
Systtem gives following error:
"for zero-ui  no matching manifest for linux/arm64/v8 in the manifest list entries"
Is it possiblee to run on arm?

Anyway great project and thumbs up!

**Reply by dec0dOS** [reply]
Hi!

Currently, there is no linux/arm64/v8 prebuilt image because Alpine does not provide a ZeroTier package for ARM. I'd try another option to build it via CI.

For now, there is a quick fix. You could install ZeroTier Controller on your host system, but continue to use ZeroUI in Docker.

Here are the steps:
1. Install ZeroTier Controller on your host device OS
```sh
curl -s https://install.zerotier.com | sudo bash
```
2. Allow access to controller from 0.0.0.0
```sh
echo "{\"settings\": {\"portMappingEnabled\": true,\"softwareUpdate\": \"disable\",\"allowManagementFrom\": [\"0.0.0.0/0\"]}}" > /var/lib/zerotier-one/local.conf
```
Optional: Please block TCP 9993 from WAN for security reasons, but not UDP.
3. Change ZeroTier controller URL in docker-compose.yml to http://127.0.0.1:9993
```docker-compose.yml
    environment:
      - ZU_CONTROLLER_ENDPOINT=http://127.0.0.1:9993/
```

Do you try to use ZeroUI on Raspberry Pi?

Please let me know if it works for you! :)

**Reply by AxlRose1976** [reply]
Thanks for your help.
But I can not even pull the image zero-ui, because of the error mentioned above.
I have the controller installed in my host.
Everything works fine so far, but I want to setup my own controller with zero-ui.
My machine is a cloud based VPS server based on Ampere A1 hardware. But this is also an arm base.

**Reply by dec0dOS** [reply]
Oh, I forgot to add that you should also remove `zerotier` from `docker-compose.yml` and the instructions to build it. Here are the updated steps:
1. Install ZeroTier Controller on your host device OS
```sh
curl -s https://install.zerotier.com | sudo bash
```
2. Clone the repo:
`git clone https://github.com/dec0dOS/zero-ui`
3. Replace `docker-compose.yml` in `zero-ui` folder with the following:
```docker-compose.yml
version: "3.9"

services:
  zero-ui:
    image: dec0dos/zero-ui:latest
    container_name: zu-main
    build:
      context: .
      dockerfile: ./docker/zero-ui/Dockerfile
    restart: unless-stopped
    volumes:
      - /var/lib/zerotier-one:/var/lib/zerotier-one
      - zero-ui_data:/app/backend/data
    environment:
      - ZU_CONTROLLER_ENDPOINT=http://127.0.0.1:9993/
      - ZU_SECURE_HEADERS=false
      - ZU_DEFAULT_USERNAME=admin
      - ZU_DEFAULT_PASSWORD=zero-ui
    network_mode: host

volumes:
  zero-ui_data:
```
4. Build Docker images for ARM
`docker-compose build`
5. Start the ZeroUI 
`docker-compose up -d`

I'd provide the installation script to automate this setup for ARM-based devices.
Thanks for your feedback, It helps to create the requirements for future development!

**Comment by AxlRose1976** [comment]
I followed your guide above. The container of zero-ui ist up and running. But unser which address  and Port can I reach the UI? I must say that I am a beginner in Linux and Docker...

**Reply by dec0dOS** [reply]
Navigate to http://YOURIP:4000.
This setup is without automated HTTPS, as I understand, there are already some services up and running on your machine, so the better way to set the HTTPS is by running the reverse proxy like NGINX on your host machine OS. 

If you could not navigate to this address, check out the logs (`docker-compose logs`) and your firewall settings.

You may also use the SSH port tunneling to get access through a firewall and establish a secure encrypted connection.
`ssh -N -L 4000:localhost:4000 sshdomain.example -p 22`
(and then navigate to http://localhost:4000)

**Reply by dec0dOS** [reply]
Hey, @AxlRose1976, did you succeed with the installation?

**Reply by AxlRose1976** [reply]
Hi dec0dOS, didn´t get this working. I installed zerotier an your docker images. Then I tried to access the UI on port 4000, but this didn´t work. NNo success with the Zerotier IP and no success from the public IP of the cloud machine. Access to Cockpit and Yacht is possible. After reading ome hints, itt could be the Firewall that is blocking access. I have to look deeper into  it...

**Reply by AxlRose1976** [reply]
Started from the beginnin,but now I gett following error when building the image:

error An unexpected error occurred: "https://registry.yarnpkg.com/@fontsource/roboto/-/roboto-4.4.5.tgz: getaddrinfo EAI_AGAIN registry.yarnpkg.com".

and the build fails...

**Reply by dec0dOS** [reply]
That is really strange. I have to test the builds on ARM.

**Reply by dec0dOS** [reply]
Hey, @AxlRose1976!
Check out 1.0.20 release with ARM support.
No more is needed to build the image yourself.

Let me know if it works for you.

**Comment by devank28** [comment]
hi,
i followed the steps and i am able to install zero-ui but on my ubuntu setup i am able to hit http:llocalhost:4000 and login but right after that the page disappear and nothing appears on the page, its all blank, i have triest remote machines and different browser, all are experiencing same behaviour, can u please provide a fix !!
